### ISSUE TITLE
How to add browsers from /Users/<USER>/Applications

### ISSUE THREAD

**Issue by floek** [will-stone/browserosaurus#399]
Hi,

my browserosaurus is not showing the browsers I've installed /Users/USER/Applications. Any hints how to achieve this?

Thanx,

floek

**Comment by will-stone** [comment]
Hi Floek,

Thanks for using Browserosaurus! B uses the package `app-exists` to find browsers. It's currently limited to `/Applications` and `~/Applications`: https://github.com/sindresorhus/app-exists/pull/4/files#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R9

Please log an issue or PR with the [`app-exists`](https://github.com/sindresorhus/app-exists) package repo if you'd like to see this support expanded to more folders.

Thanks,

Will.

**Comment by floek** [comment]
Hi Will,

okay. The browsers live under ~/Applications. So why is my browser list empty (I have only safari)

Floek

**Comment by will-stone** [comment]
Please ensure that Spotlight's indexing for applications is enabled in the Spotlight settings, as this is required for Browserosaurus to find your installed browsers.

**Comment by floek** [comment]
I have no restrictions there. When I search in Spotlight, the browsers are found.

**Comment by will-stone** [comment]
Just to be sure:

- Go to System Preferences > Spotlight.
- Untick `Applications` and tick it again.
- Wait an arbitrary amount of time, say a couple of minutes.
- Restart Browserosaurus.

**Comment by floek** [comment]
<img width="461" alt="image" src="https://user-images.githubusercontent.com/5064958/118613762-58b7a900-b7bf-11eb-8c24-81bb4125075e.png">
 Still no luck :-(

**Comment by will-stone** [comment]
Again, just to make sure, do they show in the settings screen?

<img width="465" alt="image" src="https://user-images.githubusercontent.com/654103/118614266-784ad380-b7b7-11eb-9fb8-97eed6a3439b.png">

**Comment by floek** [comment]
Thank you, but I've checked this:
<img width="463" alt="image" src="https://user-images.githubusercontent.com/5064958/118614475-16429c00-b7c0-11eb-9805-7bc52e7b9f3b.png">

**Comment by floek** [comment]
```
~> ls -d1 ~/Applications/{Firefox,"Google Chrome"}.app
/Users/user/Applications/Firefox.app
/Users/user/Applications/Google Chrome.app
```

**Comment by will-stone** [comment]
Does this work?

```
mdfind "kMDItemContentType == 'com.apple.application-bundle' && kMDItemCFBundleIdentifier == 'org.mozilla.firefox'"
```

**Comment by floek** [comment]
Yes:
```
~> mdfind "kMDItemContentType == 'com.apple.application-bundle' && kMDItemCFBundleIdentifier == 'org.mozilla.firefox'"
/Users/user/Applications/Firefox.app
/Users/user/Applications/FirefoxESR.app
fnd@mbp570:~>
```

**Comment by will-stone** [comment]
Please try this one:

```
mdfind "kMDItemContentType == 'com.apple.application-bundle' && kMDItemCFBundleIdentifier == 'org.mozilla.firefox' --onlyin '/applications' '~/Applications'"
```

**Comment by floek** [comment]
Also okay:

```
~> mdfind "kMDItemContentType == 'com.apple.application-bundle' && kMDItemCFBundleIdentifier == 'org.mozilla.firefox' --onlyin '/applications' '~/Applications'"
/Users/user/Applications/Firefox.app
/Users/user/Applications/FirefoxESR.app
fnd@mbp570:~>
```

**Comment by will-stone** [comment]
Well, I'm out of suggestions 🤷‍♂️

**Comment by floek** [comment]
Maybe an issue with https://github.com/sindresorhus/app-exists ? 
How can I test this library?

**Comment by floek** [comment]
Another info: I think I was already working in the past. Maybe I find an old version, where it worked.

**Comment by floek** [comment]
14.0.2 is working:
<img width="463" alt="image" src="https://user-images.githubusercontent.com/5064958/118617961-6e2ed200-b7c3-11eb-8dc4-e3fb090a09f2.png">

**Comment by will-stone** [comment]
Which version of macOS are you using?

**Comment by floek** [comment]
Hi Will,

it’s 11.3.1

Cheers,

Floek

**Comment by will-stone** [comment]
Well, it's very strange. because if `app-exists` comes back with anything other than `undefined` then it should show the browser. One last thing, have you tried scrolling the settings / making B bigger, just to make sure the other browsers aren't hiding below the fold?

**Comment by WorkInProgress-Development** [comment]
Just chipping in to add that this issue is also happening to me.
macOS 10.15.7
14.1.0 works fine but 14.1.2 and 14.1.3 do not.
I've checked that the browsers aren't hiding below the fold.

**Comment by WorkInProgress-Development** [comment]
When I change the path in ```app-exists``` from ~/Applications to the absolute path ```/Users/myusername/Applications```, B finds the browsers. This appears to be as node does not expand tildes by default (nodejs/node#684).
I'll make an issue referencing sindresorhus/app-exists#4 in the latter repo.

**Comment by will-stone** [comment]
Nice one @WorkInProgress-Development 😄 Thank you.

**Comment by WorkInProgress-Development** [comment]
Pleasure, and thanks for a **very** useful tool!

**Comment by will-stone** [comment]
Fixed in v14.1.4

**Comment by Tragio** [comment]
Hello @will-stone 🙂  I'm using Browserosaurus for months and one of these last updates made most of the browsers disappeared. I already updated to the latest version, refreshed the Browserosaurus, and restarted it. 🤔
